types/bool.go

```diff
@@ -0,0 +1,40 @@
+package types
+
+type (
+	Bool bool
+)
+
+var True = Bool(true)
+var False = Bool(false)
+
+func NewBool(b bool) Bool    { return Bool(b) }
+func (_ Bool) Kind() Kind    { return KindBool }
+func (b Bool) IsTrue() bool  { return bool(b) }
+func (b Bool) IsFalse() bool { return !bool(b) }
+func (b Bool) ToBool() Bool  { return b }
+func (b Bool) ToNumber() Number {
+	if b {
+		return NewNumber(1)
+	}
+	return NewNumber(+0)
+}
+
+func (b Bool) ToString() String {
+	if b {
+		return NewString("true")
+	}
+	return NewString("false")
+}
+
+func (b Bool) ToPrimitive(hint Kind) (Value, error) {
+	return b, nil
+}
+
+func (b Bool) ToObject() (Object, error) {
+	panic("not implemented")
```

Comment:
Isnt everything an object on JS ? =P

Reply:
no =P

Reply:
To implement `ToObject` requires the implementation of `wrapper objects` for primitive values.